tools/test

```diff
@@ -87,9 +87,33 @@ iotjs /usr/lib/node_modules/tape/bin/tape.js
   fi
 fi
 
-for pattern in $patterns; do
-  if ! test -z "$pattern"; then
-    printf "# $patternPrefix$pattern\n"
-    $cmd $pre_scripts $coverage $timeout "$patternPrefix$pattern" | $reporter
-  fi
-done
+function run_device(){
+  # generate the coverage data for one directory and than the next data will covery it.
+  for pattern in $patterns; do
+    if ! test -z "$pattern"; then
+      printf "# $patternPrefix$pattern\n"
+      $cmd $pre_scripts $coverage $timeout "$patternPrefix$pattern" | $reporter
+    fi
+  done
+}
+
+function run_local(){
+  # run all case and generate the coverage data file
+  test_list=""
+  printf "The below test list will be executed:\n"
+  while read -r pattern;
+  do
+    if ! test -z "$pattern" && ! [[ "$pattern" =~ "#" ]]; then
+      test_list="$test_list$patternPrefix$pattern "
+      printf "> $patternPrefix$pattern\n"
+    fi
+  done<${testset}
+  printf "\n"
+  $cmd $pre_scripts $coverage $timeout ${test_list} | $reporter
```

Comment:
We'd expect similar test process management on both device and local machine. So I'd recommend separate `$cmd` execution for each test case.